indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -93,6 +93,33 @@ public static Path distributedClassPath(Path base)
   }
 
   /**
+   * Dose authenticate against a secured hadoop cluster
+   * In case of any bug fix make sure to fix the code at HdfsStorageAuthentication#authenticate as well.
+   * @param config containing the principal name and keytab path.
+   */
+  public static void authenticate(HadoopDruidIndexerConfig config)
```

Comment:
nvm, there seems to be no common place to put this stuff up. can you add the reason for code dup to javadoc as well. 
Also, both the methods have slightly diff condition checks already. Making them identical would help in propagating any code changes.